### PR TITLE
Adding https server implementation

### PR DESCRIPTION
**Pull request by dmarro89**
Adding implementation for https server.

### CODE REVIEW

File: server/configuration_test.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import (
+	"os"
+	"testing"
+
+	"gotest.tools/assert"
+)
+
+func TestNewConfiguration(t *testing.T) {
+	// Set environment variables
+	os.Setenv(DARE_PORT, "8080")
```
Comment (arl) [inline server/configuration_test.go:12]: `t.Setenv` is preferable in tests since it takes care of setting back the variables once the test ends.

Comment (dmarro89) [reply server/configuration_test.go:12]: ok thanks for the comment!

**Comment by dmarro89** [comment]
> Except the comment to use a constant as a multiplicator for the initial start (sleep at start), no further comments from me (based on reading the code)

Could you provide some example code ?

File: server/server_test.go [hunk 2]
```diff
@@ -1,108 +1,103 @@
...
+	mux := http.NewServeMux()
+	server.dareServer.(*MockDareServer).On("CreateMux").Return(mux)
 
-	setData := map[string]interface{}{"testKey": "testValue"}
-	setDataJSON, _ := json.Marshal(setData)
-	setRequest, _ := http.NewRequest("POST", "/set", bytes.NewBuffer(setDataJSON))
-	setResponse := httptest.NewRecorder()
+	// Start server
+	go server.Start()
+	time.Sleep(3 * time.Second) // Give it time to start
```
[the remark this answers is not in the document]
Comment (vdmitriyev) [reply server/server_test.go:57]: Here is an example (later on this should be moved to the config file):

```go
const sleepOnStart int = 3

time.Sleep(sleepOnStart * time.Second) // Give it time to start
```

[the remark this answers is not in the document]
Comment (vdmitriyev) [reply server/server_test.go]: Added this line to the tests and pushed changes to this pull request

**Comment by vdmitriyev** [comment]
Make TLS config names in sync with documentation

**Pull request merged**